fluent/fluent.go

```diff
@@ -297,6 +301,12 @@ func (f *Fluent) send() error {
 
 	var err error
 	if len(f.pending) > 0 {
+		t := f.Config.WriteTimeout
+		if time.Duration(0) < t {
+			f.conn.SetWriteDeadline(time.Now().Add(t))
+		} else {
+			f.conn.SetWriteDeadline(time.Time{})
```

Comment:
Since `f.Config.WriteTimeout` is an exported field, I made it possible to change the behavior from time-out to not-time-out, anytime when 0 is set. Aside from whether it has a substantial meaning, I thought it was a behavior close to expectation as a user.

```diff
diff --git a/fluent-logger/main.go b/go/fluent-logger/main.go
index eb5614a..63e2ad4 100644
--- a/go/fluent-logger/main.go
+++ b/go/fluent-logger/main.go
@@ -26,6 +26,7 @@ func main() {
                e := logger.Post(tag, data)
                if e != nil {
                        log.Println("Error while posting log: ", e)
+                       logger.Config.WriteTimeout = 0
                } else if (i % 10000) == 0 {
                        log.Println("Success to post log: ", i)
                }
```